core/Visualization/Sparkline.php

```diff
@@ -12,168 +12,155 @@
 use Piwik\Common;
 use Piwik\Piwik;
 use Piwik\View\ViewInterface;
-use Sparkline_Line;
-
-/**
- * @see libs/sparkline/lib/Sparkline_Line.php
- * @link http://sparkline.org
- */
-require_once PIWIK_INCLUDE_PATH . '/libs/sparkline/lib/Sparkline_Line.php';
 
 /**
  * Renders a sparkline image given a PHP data array.
  * Using the Sparkline PHP Graphing Library sparkline.org
  */
 class Sparkline implements ViewInterface
 {
-    const DEFAULT_WIDTH = 100;
-    const DEFAULT_HEIGHT = 25;
+    const DEFAULT_WIDTH = 200;
+    const DEFAULT_HEIGHT = 50;
 
     public static $enableSparklineImages = true;
 
-    private static $colorNames = array('backgroundColor', 'lineColor', 'minPointColor', 'lastPointColor', 'maxPointColor');
-    private $values = array();
-
     /**
      * Width of the sparkline
      * @var int
      */
     protected $_width = self::DEFAULT_WIDTH;
-
     /**
      * Height of sparkline
      * @var int
      */
     protected $_height = self::DEFAULT_HEIGHT;
+    private $values = array();
+    /**
+     * @var \Davaxi\Sparkline
+     */
+    private $sparkline;
 
     /**
      * Array with format: array( x, y, z, ... )
      * @param array $data
      */
-    public function setValues($data)
-    {
+    public function setValues($data) {
         $this->values = $data;
     }
 
-    /**
-     * Sets the height of the sparkline
-     * @param int $height
-     */
-    public function setHeight($height)
-    {
-        if (!is_numeric($height) || $height <= 0) {
-            return;
+    public function main() {
+
+        $sparkline = new \Davaxi\Sparkline();
+
+        $seconds = Piwik::translate('Intl_NSecondsShort');
+        $toRemove = array('%', str_replace('%s', '', $seconds));
+        $values = [];
+        foreach ($this->values as $value) {
+            // 50% and 50s should be plotted as 50
+            $value = str_replace($toRemove, '', $value);
+            // replace localized decimal separator
+            $value = str_replace(',', '.', $value);
```

Comment:
That peace of code is a bit outdated and won't work for all languages. Some languages use different characters to present decimals or percent values.
At first I thought about adding a `$this->requestConfig->request_parameters_to_modify['format_metrics'] = 0;` to the `render()` method of the Sparkline ViewDataTable, so the numbers aren't formatted at all, as that would remove any possible custom format, but that would result in decimal numbers instead of percent values (e.g. 0.53 instead of 53%), which then wouldn't show up correctly in the sparkline. Can't think of a good solution to handle the decimal percent values so the show up better.
Alternatively we could create a new method in NumberFormatter to convert a formatted number back to english format. Like https://github.com/matomo-org/matomo/blob/3.x-dev/core/NumberFormatter.php#L296-L306 does, but with flipped and reversed replacement values. But that would be more like a workaround.

Reply:
To be honest, I haven't thought a lot about it, but kept it the same [as it was before](https://github.com/matomo-org/matomo/blob/3.x-dev/core/Visualization/Sparkline.php#L116)

Is it guaranteed that the decimal value will always have two decimal points? Then one could simply multiply it by 100.